storage/reads/datatypes/storage_common.proto

```diff
@@ -208,6 +214,18 @@ message TimestampRange {
   int64 end = 2;
 }
 
+// TagsRequest is the request message for the Storage.TagKeys and Storage.TagValues
+message TagsRequest {
+  google.protobuf.Any tags_source = 1 [(gogoproto.customname) = "TagsSource"];
+  TimestampRange range = 2 [(gogoproto.nullable) = false];
+  Predicate predicate = 3;
+}
+
+// Response message for Storage.TagKeys and Storage.TagValues.
+message TagsResponse {
```

Comment:
This should be renamed to something more generic, like `StringValuesResponse` given it represents the generic `StringIterator` and its associated merge operation.